client/components/button/README.md

```diff
@@ -17,12 +18,53 @@ export default function RockOnButton() {
 }
 ```
 
-#### Props
+### Props
 
-The following props are used to control the display of the component. Any additional props will be passed to the rendered `<a />` or `<button />` element. The presence of a `href` prop determines whether an anchor element is rendered instead of a button.
+Name | Type | Default | Description
+--- | --- | --- | ---
+`compact` | `bool` | false | Decreases the size of the button
+`primary` | `bool` | false | Provides extra visual weight and identifies the primary action in a set of buttons
+`borderless` | `bool` | false | Renders a button without borders
+`scary` | `bool` | false | Indicates a dangerous or potentially negative action
+`busy` | `bool` | false | Indicates activity while a background action is being performed
+`href` | `string` | null | If provided, renders `a` instead of `button`
 
-* `compact`: (bool) whether the button is compact or not.
-* `primary`: (bool) whether the button is styled as a primary button.
-* `scary`: (bool) whether the button has modified styling to warn users (delete, remove, etc).
-* `href`: (string) if this property is added, it will use an `a` rather than a `button` element.
-* `borderless`: (bool) whether button should be displayed without borders
+### Button types
+
+* **Primary**: Use to highlight the most important actions in any experience. Don’t use more than one primary button in a section or screen to avoid overwhelming customers.
+* **Secondary**: Used most in the interface. Only use another style if a button requires more or less visual weight.
+* **Button with icon**: When words are not enough, icons can be used in buttons to better communicate what the button does.
+* **Scary**: Use when the action will delete customer data or be otherwise difficult to recover from. Destructive buttons should trigger a confirmation dialog before the action is completed. Be thoughtful about using destructive buttons because they can feel stressful for customers.
+* **Borderless**: Use for less important or less commonly used actions since they’re less prominent.
+* **Busy**: Use when a button has been pressed and the associated action is in progress.
+
+#### Icon buttons
```

Comment:
@iamtakashi Hey Takashi, I'm updating the button component readme. Do you mind reading the "Icon buttons" section and telling be if it's accurate? It looks like we might not need the `button-text` class.